### ISSUE TITLE
Step 1 - Unfilled DEM not creating additional products

### ISSUE THREAD

**Issue by kennethelsner** [ARS-SWRC/agwa#5]
I'm not using the tutorial. Running on real world data.

I added an unfilled DEM to step 1 and ran the tool. According to the AGWA Pro, Mountain Fire Tutorial, step 14.4, if the the DEM already filled is **unchecked**  the DEM parameter is assumed to be an unfilled DEM. As a result, AGWA will create and store the filled DEM, flow direction, flow accumluation, flow length up, slope, and aspect rasters **automatically**

Tool says 'Step 1 - Setup AGWA Workspace Completed'.  (ss1)
![AGWAPro_Step1_issue1_ss1](https://github.com/user-attachments/assets/f1a7b714-a19a-4424-bb20-61596c46ae38)

Metaworkspace table says FilledDEM, FD, FA, FLUP, Slope, Aspect are located in the fgdb. (ss2)
![AGWAPro_Step1_issue1_ss2](https://github.com/user-attachments/assets/819e4199-e986-4dc5-9670-58c97e3abefd)

When I look in the fgdb after refreshing, no data is present. (ss3)
![AGWAPro_Step1_issue1_ss3](https://github.com/user-attachments/assets/06a5ee30-15e9-47aa-b199-3e3e86715e34)

Do I need to produce the Filled DEM, FD, FA, FLUP, Slope, Aspect manually? am I missing something obvious? or is something not working quite right?

**Comment by isburns** [comment]
Hi Kenny, it looks like these are being created and the path is being recorded correctly in the `Metaworkspace` table, but it's not saving them where it should. Take a closer look and you'll see it's saving them as `.tif` in a folder within the file geodatabase. I'm not sure if the intent here is to save them as a file geodatabase raster in that database, or as a `.tif` in the folder that the file geodatabase resides.

I'm not sure this actually breaks anything because you can actually load those into a map, you just have to figure out how to navigate to them or do it in code (which AGWA is doing). However, even though it works doesn't mean it's correct, I think this should be addressed. In the screenshot below, I clicked into the navigation bar and pasted the path and was then able to see and access the rasters. See if that gets you moving forward for now.
![input_rasters](https://github.com/user-attachments/assets/a32d27fa-cfb0-4a53-adda-3236c8e91358)

**Comment by kennethelsner** [comment]
Without understanding of how to navigate to the .tifs, I created them manually and was able to run through AGWA until I ran into the Join and View Sim Results error. 

It makes sense that the metaWorkspace table points to the .tifs, my brain just didn't connect the dots. 

It would make sense if the rasters are 'added to the map' after they are generated. 

Thanks for responding!

**Comment by haiyan-Wei** [comment]
@kennethelsner 
I worked on this last week and updated Step 1. The new interface indicates what AGWA is asking for and what it can create for users. It also provides an option for users to choose a directory to save the rasters created by AGWA. 

The tutorial is updated accordingly. 

If you have time and would like to give it a try, please let me know how it works out for you. I would really appreciate your feedback.

**Comment by kennethelsner** [comment]
The interface for step 1 makes A LOT more sense. Thanks for that.
With the following inputs
![image](https://github.com/user-attachments/assets/4b0b0ddd-060f-42b8-9029-aa5a86c66cca)

I received this error
![image](https://github.com/user-attachments/assets/7a389322-93e3-4653-8873-09922482cc92)

These are the python files I added to the 'src folder'
![image](https://github.com/user-attachments/assets/fa1439ba-0f3c-4f2c-98df-084aa225b462)

**Comment by haiyan-Wei** [comment]
@kennethelsner 
It’s great to hear that the new interface makes sense to you, and thanks again for always including a screenshot—it’s always helpful!

Regarding the error you mentioned, I wonder if you’ve updated the file "code_setup_agwa_workspace.py"? The updated version should be around 8KB, but from your screenshot, it appears to be 6KB and it was not highlighted by you.

**Comment by kennethelsner** [comment]
@haiyan-Wei  Yep. My fault. Updating the 'code_setup_agwa_workspace' fixed the issue. Thanks for bearing with me as I bumble through this.

So the functionality seems to be working as intended! Yea!

Side comment for step 1, the aspect generated by AGWA has some 'griding'. 
![image](https://github.com/user-attachments/assets/9e11f513-be0e-4991-a015-b12a938b4923)

I have seen this 'artifact' in the past due to using an improper resampling technique on raster. ArcGIS Pro, by default will use bilinear interpolation when creating rasters, when continuous data like elevation products should be using cubic convolution. It certainly effects the delineation and disc output (see block nature of disc and delineation in screenshot).

**Comment by haiyan-Wei** [comment]
@kennethelsner Awesome! 
Thanks for bringing the gridding effect to my attention. You’re right—we should be using the cubic option for interpolation. I’ll make sure to fix that!

**Comment by haiyan-Wei** [comment]
A quick update: "cubic" interpolation method is now used in Step 1 -  fixed and committed on Sept 6.
